apps/meteor/app/api/server/v1/users.ts

```diff
@@ -555,6 +559,129 @@ API.v1.addRoute(
 	},
 );
 
+API.v1.addRoute(
+	'users.listByStatus',
+	{
+		authRequired: true,
+		validateParams: isUsersListStatusProps,
+		permissionsRequired: ['view-d-room', 'view-outside-room'],
+	},
+	{
+		async get() {
+			const { offset, count } = await getPaginationItems(this.queryParams);
+			const { sort, fields } = await this.parseJsonQuery();
+			const { status, roles, searchTerm } = this.queryParams;
+
+			const projection = {
+				name: 1,
+				username: 1,
+				emails: 1,
+				roles: 1,
+				status: 1,
+				active: 1,
+				avatarETag: 1,
+				lastLogin: 1,
+				type: 1,
+				reason: 0,
+				...fields,
+			};
+
+			const actualSort: Record<string, 1 | -1> = sort || { username: 1 };
+
+			if (sort?.status) {
+				actualSort.active = sort.status;
+			}
+
+			if (sort?.name) {
+				actualSort.nameInsensitive = sort.name;
+			}
+
+			let match: Filter<IUser>;
+
+			switch (status) {
+				case 'active':
+					match = {
+						active: true,
+						lastLogin: { $exists: true },
+					};
+					break;
+				case 'all':
+					match = {};
+					break;
+				case 'deactivated':
+					match = {
+						active: false,
+						lastLogin: { $exists: true },
+					};
+					break;
+				case 'pending':
+					match = {
+						lastLogin: { $exists: false },
+						type: { $nin: ['bot', 'app'] },
+					};
+					projection.reason = 1;
+					break;
+				default:
+					throw new Meteor.Error('invalid-params', 'Invalid status parameter');
+			}
+
+			const canSeeAllUserInfo = await hasPermissionAsync(this.userId, 'view-full-other-user-info');
+
+			match = {
+				...match,
+				$or: [
+					...(canSeeAllUserInfo ? [{ 'emails.address': { $regex: escapeRegExp(searchTerm), $options: 'i' } }] : []),
+					{ username: { $regex: escapeRegExp(searchTerm), $options: 'i' } },
+					{ name: { $regex: escapeRegExp(searchTerm), $options: 'i' } },
+				],
+			};
+
+			if (roles?.length && !roles.includes('all')) {
+				match = {
+					...match,
+					roles: { $in: roles },
+				};
```

Comment:
```suggestion
				match.roles = { $in: roles };
```
This should work, right? Looks cleaner to me